### DISCUSSION TITLE
Segmentation Fault in Level-2-ImproPhe

### DISCUSSION THREAD

**Discussion by skoenig-eo** [davidfrantz/force#309] [Q&A]
Hi all,

I have a question regarding the Level-2-ImproPhe module in FORCE. I have a bunch of Level-2 Landsat data at 30 m spatial resolution the resolution of which I'd like to increase to 10 m using ImproPhe and Level-2 Sentinel-2 data. I've struggeled with this for some time, because regardless of the options I apply, I encounter a segmentation fault error right after starting the process (tried various different combinations of options). This is the output in the console:

```
number of processing units: 10
 (active tiles: 1, chunks per tile: 10)
________________________________________
Progress:                          0.00%
Time for I/C/O:        not available yet
ETA:                   not available yet
________________________________________
                   input compute  output
Processing unit:       0      -1      -2
Tile X-ID:             0      -1      -1
Tile Y-ID:             0      -1      -1
Chunk ID:              0      -1      -1
Threads:               1       1       1
Time (sec):            0       0       0
Segmentation fault
```

As far as I see it, I've done everything according to the recommendations for this task (see the attached param file). It might be because I'm using the WSL Ubuntu, but all other FORCE modules I've used so far worked smoothly. Is there any property to the ImproPhe process that may induce errors like this, or could anyone of you imagine why a segmentation fault may be incuded in general during ImproPhe? 

As alyways, any help is appreciated. Thanks a lot in advance!

Cheers,
Simon

[lx_improphe.txt](https://github.com/davidfrantz/force/files/12865875/lx_improphe.txt)

P.S.: this is just a basic overview of the applied parameters, I've tried out a lot of options in the ImproPhe and parallel processing sections.

**Comment by davidfrantz** [comment]
Hi Simon,

a segfault should generally not happen. Are you using the latest version?

From your parameter file, I don't see major issues.

Only two small things, which should not explain the segfault:

- For this specific module, I would suggest to use the same output folder as the input.

- I'd also suggest to work with one year of data at a time (DATE_RANGE).

Have you tried running FORCE in debug mode?

David
